coil-base/src/main/java/coil/decode/ImageSource.kt

```diff
@@ -133,14 +141,18 @@ sealed class ImageSource : Closeable {
      * Return a [File] containing this [ImageSource]'s data.
      * If this image source is backed by a [BufferedSource], a temporary file containing this
      * [ImageSource]'s data will be created.
+     *
+     * This method is only safe when used on FileSystem.SYSTEM.
      */
-    abstract fun file(): File
+    abstract fun file(): Path
 
     /**
      * Return a [File] containing this [ImageSource]'s data if one has already been created.
      * Else, return 'null'.
      */
-    abstract fun fileOrNull(): File?
+    abstract fun fileOrNull(): Path?
```

Comment:
I think we also need a `val fileSystem: FileSystem` for `ImageSource`, right? We can always return `FileSystem.SYSTEM` for image sources not backed by a file.